graphs/bipartite.py

```diff
@@ -0,0 +1,59 @@
+import sys
+
+"""
+An undirected graph is called bipartite if its vertices can be split into two parts such that each edge of the
+graph joins to vertices from different parts. Bipartite graphs arise naturally in applications where a graph
+is used to model connections between objects of two different types (say, boys and girls; or students and
+dormitories).
+An alternative definition is the following: a graph is bipartite if its vertices can be colored with two colors
+(say, black and white) such that the endpoints of each edge have different colors.
+"""
+
+def bipartite(partition, adj):
```

Comment:
```suggestion
def bipartite(partition: list, adj: list) -> int:
```
https://docs.python.org/3/library/typing.html

Also needed: https://docs.python.org/3/library/doctest.html